apps/req.c

```diff
@@ -1569,6 +1575,10 @@ static EVP_PKEY_CTX *set_keygen_ctx(const char *gstr,
         }
         if (keytype == NULL) {
             keytype = EVP_PKEY_get0_type_name(param);
+            if (keytype == NULL) {
+                BIO_puts(bio_err, "Unable to determine key type\n");
```

Comment:
Looks like a memory leak will happen here now? (param)

Reply:
Good point.  I don't think that this path is actually possible but better to be safe and check for it.